superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterSets/utils.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import shortid from 'shortid';
+import { t } from '@superset-ui/core';
+
+export const generateFiltersSetId = () => `FILTERS_SET-${shortid.generate()}`;
+
+export const getFilterValueForDisplay = (
+  value?: string[] | null | string | number | object,
+): string => {
+  if (value === null || value === undefined) {
+    return '';
+  }
+  if (typeof value === 'string' || typeof value === 'number') {
+    return `${value}`;
+  }
+  if (Array.isArray(value)) {
+    return value.join(', ');
+  }
+  if (typeof value === 'object') {
+    return JSON.stringify(value);
+  }
+  return t('Unknown value');
+};
```

Comment:
It would be nice to use the same formatter here as in the filter indicator (perhaps implement this there?): https://github.com/apache/superset/blob/49eeab6f55e9e1f35b8a46e2906fe42d1cdf1f72/superset-frontend/src/dashboard/components/FiltersBadge/DetailsPanel.tsx#L50-L58